iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/PrivilegedUserCheck.java

```diff
@@ -84,27 +86,36 @@ private void handle(CheckContext ctx, DockerImage dockerImage) {
     if(!isLastDockerImageInFile(dockerImage)) {
       return;
     }
-    String imageName = getImageName(dockerImage.from());
+
+    getLastUser(dockerImage).ifPresentOrElse(
+      lastUserInstruction -> checkLastUserInstruction(ctx, lastUserInstruction), 
+      () -> checkLastImageName(ctx, dockerImage.from())
+    );
+  }
+
+  private static void checkLastUserInstruction(CheckContext ctx, UserInstruction userInstruction) {
+    for(Argument user: userInstruction.arguments()) {
+      String userName = ArgumentUtils.resolve(user).value();
+      if (UNSAFE_USERS.contains(userName)) {
+        ctx.reportIssue(userInstruction, String.format(MESSAGE_ROOT_USER, userName));
+      }
+    }
+  }
```

Comment:
It's not really working as before, as it is checking each USER argument individually.
So currently, `USER my user root` would raise an issue, while it was not before.

From what I read, this is apparently possible in Windows system to have spaces in username :D `John Smith` is a valid user name (although this apparently cause a lot of trouble for poor user who have put spaces in their username, especially for dev trying to install some tool/package which use %USERNAME% env variable).
I was able to create one in my system :
![image](https://user-images.githubusercontent.com/110470341/223724439-76b7096f-d2fe-4e31-a474-6b77bf01f800.png)


Still our current implementation or Argument does not really permit this, as we don't store whitespace. Probably this could be fixed in another ticket.
This also mean that we would still need a method like `resolveAndMerge()` to merge an `Argument` list as a single string.

I hope that most dev/admin does not use user with whitespace in it, still this is possible.

Reply:
Nice catch and thank you for testing. I will adapt the check impl.